sdk/storage/src/common/storage_error.cpp

```diff
@@ -4,92 +4,120 @@
+            if (std::strcmp(node.Name, "Error") == 0)
+            {
+              path.emplace_back(XmlTagName::c_Error);
+            }
+            else if (std::strcmp(node.Name, "Code") == 0)
+            {
+              path.emplace_back(XmlTagName::c_Code);
+            }
+            else if (std::strcmp(node.Name, "Message") == 0)
+            {
+              path.emplace_back(XmlTagName::c_Message);
+            }
+          }
+          else if (node.Type == XmlNodeType::Text)
+          {
+            if (path.size() == 2 && path[0] == XmlTagName::c_Error && path[1] == XmlTagName::c_Code)
+            {
+              errorCode = node.Value;
+            }
+            else if (
+                path.size() == 2 && path[0] == XmlTagName::c_Error
+                && path[1] == XmlTagName::c_Message)
+            {
+              message = node.Value;
+            }
+          }
         }
       }
-      else if (node.Type == XmlNodeType::StartTag)
+      else if (response->GetHeaders().at("Content-Type").find("html") != std::string::npos)
       {
-        if (std::strcmp(node.Name, "Error") == 0)
-        {
-          path.emplace_back(XmlTagName::c_Error);
-        }
-        else if (std::strcmp(node.Name, "Code") == 0)
-        {
-          path.emplace_back(XmlTagName::c_Code);
-        }
-        else if (std::strcmp(node.Name, "Message") == 0)
-        {
-          path.emplace_back(XmlTagName::c_Message);
-        }
-        else
-        {
-          path.emplace_back(XmlTagName::c_Details);
-        }
+        // TODO: add a refined message parsed from result.
+        message = std::string(bodyBuffer->begin(), bodyBuffer->end());
       }
-      else if (node.Type == XmlNodeType::Text)
+      else
       {
-        if (path.size() == 2 && path[0] == XmlTagName::c_Error && path[1] == XmlTagName::c_Code)
-        {
-          code = node.Value;
-        }
-        else if (
-            path.size() == 2 && path[0] == XmlTagName::c_Error && path[1] == XmlTagName::c_Message)
-        {
-          message = node.Value;
-        }
-        else if (
-            path.size() == 2 && path[0] == XmlTagName::c_Error && path[1] == XmlTagName::c_Details)
-        {
-          details[node.Name] = node.Value;
-        }
+        // TODO: add a refined message parsed from result.
+        message = std::string(bodyBuffer->begin(), bodyBuffer->end());
       }
     }
-    StorageError result = StorageError(message, code);
-
-    if (response.GetHeaders().find("x-ms-request-id") != response.GetHeaders().end())
-    {
-      result.RequestId = response.GetHeaders().at("x-ms-request-id");
-    }
-    result.Details = std::move(details);
 
+    StorageError result = StorageError(
+        "Http Status Code: "
+        + std::to_string(static_cast<std::underlying_type<Azure::Core::Http::HttpStatusCode>::type>(
+            httpStatusCode))
+        + ", Reason Phrase: " + reasonPhrase + ", Error Message: " + message + " Request ID:\n"
```

Comment:
I fee like we don't need to print those `Http Status Code` or `Reason Phrase` because everybody knows that.

```
404 Not Found
The specified blob does not exist.
Request ID: 2B842DAD-5411-419E-BE60-AADEF1EFDAE9
```

```
Http Status Code: 404, Reason Phrase: Not Found, Error Message: The specified blob does not exist. 
Request ID:
2B842DAD-5411-419E-BE60-AADEF1EFDAE9
```

The former is clearer, isn't it?

Reply:
You are making an excellent point. I'll change the message format.